File: java-version/src/main/java/bernard/tatin/procfs/ProcessID.java
package bernard.tatin.procfs;


public class ProcessID {
    private static long pid = -1;

    public static long getPID() {
        if (pid == -1) {
            String processName =
                    java.lang.management.ManagementFactory.getRuntimeMXBean().getName();
            pid = Long.parseLong(processName.split("@")[0]);
        }
        return pid;
    }


    private ProcessID() {
    }
}

File: kotlin-version/src/main/kotlin/bernard/tatin/ProcFS/ProcessID.kt
package bernard.tatin.ProcFS


object ProcessID {
    private var pid: Long = -1

    fun getPID(): Long {
        if (pid == -1) {
            val processName = java.lang.management.ManagementFactory.getRuntimeMXBean().getName()
            pid = Long.parseLong(processName.split("@")[0])
        }
        return pid
    }
}
